### PR TITLE
Embrace the module-function nature of tress

### PR DESCRIPTION
**Pull request by sindilevich**
Fixed the definitions for tress to reflect the fact that the tress module exports a function.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/astur/tress/blob/master/index.js#L167
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by sindilevich** [comment]
A template for module-function definition was taken from https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-function-d-ts.html.

File: types/tress/tress-tests.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /// <reference types="node" />
 
-import { tress } from "tress";
```
Comment (DanielRosenwasser) [inline types/tress/tress-tests.ts]: How was this ever working for anyone?

File: types/tress/tress-tests.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+import * as tress from "tress";
```
Comment (DanielRosenwasser) [inline types/tress/tress-tests.ts]: Can you change this to `import tress = require("tress");`?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/tress/tslint.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
 	"extends": "dtslint/dt.json",
 	"rules": {
-		"space-before-function-paren": false
+		"space-before-function-paren": false,
+		"strict-export-declare-modifiers": false
```
Comment (DanielRosenwasser) [inline types/tress/tslint.json]: What are you trying to suppress here?

File: types/tress/tress-tests.ts (hunk 1 above)
Comment (sindilevich) [reply types/tress/tress-tests.ts]: @DanielRosenwasser, it wasn't. But I only paid attention to the odds of my first version of tress' definition at the moment I started using it in a Typescript project.

File: types/tress/tress-tests.ts (hunk 2 above)
Comment (sindilevich) [reply types/tress/tress-tests.ts]: @DanielRosenwasser, why is it necessary? Currently, it uses Typescript/ES6 module import style. See below for another example of the same syntax.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0e02e7b65b259990ed3e9b14a16f053a6ce5a1fa/types/needle/needle-tests.ts#L1

File: types/tress/tslint.json (hunk 3 above)
Comment (sindilevich) [reply types/tress/tslint.json]: @DanielRosenwasser, without the `strict-export-declare-modifiers` suppression linting fails on code like below.
https://github.com/sindilevich/DefinitelyTyped/blob/7cc27e8a787a9c77fc242ff6131f8fd88eb0311c/types/tress/index.d.ts#L8

**Comment by RyanCavanaugh** [comment]
@sindilevich Please address comments from the code reviewers.

**Comment by RyanCavanaugh** [comment]
@DanielRosenwasser - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**